sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RefreshTableExec.scala

```diff
@@ -17,23 +17,20 @@
 
 package org.apache.spark.sql.execution.datasources.v2
 
-import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.Attribute
-import org.apache.spark.sql.connector.catalog.{Identifier, Table, TableCatalog}
+import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
 
 case class RefreshTableExec(
-    session: SparkSession,
     catalog: TableCatalog,
-    table: Table,
-    ident: Identifier) extends V2CommandExec {
+    ident: Identifier,
+    invalidateCache: () => Unit) extends V2CommandExec {
   override protected def run(): Seq[InternalRow] = {
     catalog.invalidateTable(ident)
 
     // invalidate all caches referencing the given table
     // TODO(SPARK-33437): re-cache the table itself once we support caching a DSv2 table
```

Comment:
@sunchao let's also fix this TODO in a separate PR.

Reply:
Sure. Will do that soon.